File: low-level-design/design-patterns/python/creational/abstract-factory/gcp.py
from resources import Compute,Storage,Network

class GCPCompute(Compute):
    def __init__(self)  :
        pass 
    def Start(self):
        print("GCP Compute instance started")
    def stop(self):
        print("GCP Compute instance stopped")

class GCPStorage(Storage):
    def __init__(self):
        pass
    def Create(self,bucketName:str):
        print(f"GCP Storage bucket {bucketName} created")

class GCPNetwork(Network):
    def __init__(self):
        pass 
    def Configure(self,subnetName:str):
        print(f"GCP Network subnet {subnetName} configured")

File: low-level-design/design-patterns/golang/creational/abstract-factory/abstract-factory/gcp.go
package abstractfactory

import "fmt"


func NewGCPCompute()*GCPCompute{
	return &GCPCompute{}
}

type GCPCompute struct{

}


func(c *GCPCompute) Start() error {
	fmt.Println("Starting GCP compute resources")
	return nil
}

func(c *GCPCompute) Stop() error {
	fmt.Println("Stopping GCP compute resources")
	return nil
}

func NewGCPStorage() *GCPStorage {
	return &GCPStorage{}
}

type GCPStorage struct {

}

func(s *GCPStorage) Create(bucketName string) error {
	fmt.Println("Creating GCP storage bucket:", bucketName)
	return nil
}

func NewGCPNetwork() *GCPNetwork {
	return &GCPNetwork{}
}

type GCPNetwork struct {

}

func(s *GCPNetwork) Configure(name string) error {
	fmt.Println("Configuring GCP network:", name)
	return nil
}